libs/core/langchain_core/prompts/base.py

```diff
@@ -182,6 +182,23 @@ def format(self, **kwargs: Any) -> FormatOutputType:
             prompt.format(variable1="foo")
         """
 
+    async def aformat(self, **kwargs: Any) -> FormatOutputType:
```

Comment:
Will need to think for a bit about whether we want to include aformat. Feels fairly unusual that formatting will involve network calls

Reply:
It does for the few-shot prompt template.

Reply:
The default should be delegating to sync rather than kicking off a thread.

I think we also need to provide `ainvoke` implementation here otherwise this won't be used at run time when using LCEL

Reply:
We can also raise NotImplementedError for now and decide later/never which should be the default impl. I intend to implement it in all subclasses anyway. Not making it abstract as that would be a breaking change.

> I think we also need to provide ainvoke implementation here otherwise this won't be used at run time when using LCEL

I wanted to do it in another PR to keep changes atomic.

Reply:
> We can also raise NotImplementedError for now and decide later/never which should be the default impl. 

Actually this could lead to some breaking changes when we implement other async methods to not call their sync counterpart in a thread (eg. ainvoke).